OrbitClientModel/CaptureSerializer.cpp

```diff
@@ -74,14 +75,26 @@ CaptureInfo GenerateCaptureInfo(
     }
     // Fix names/offset/module in address infos (some might only be in process):
     added_address_info->set_function_name(FunctionUtils::GetDisplayName(*function));
-    const uint64_t offset = absolute_address - FunctionUtils::GetAbsoluteAddress(*function);
+    const ModuleData* module = capture_data.FindModuleByPath(function->loaded_module_path());
+    CHECK(module != nullptr);
+    uint64_t absolute_function_address =
+        FunctionUtils::GetAbsoluteAddress(*function, *capture_data.process(), *module);
+    const uint64_t offset = absolute_address - absolute_function_address;
     added_address_info->set_offset_in_function(offset);
     added_address_info->set_module_path(function->loaded_module_path());
   }
 
-  const absl::flat_hash_map<uint64_t, FunctionStats>& functions_stats =
-      capture_data.functions_stats();
-  capture_info.mutable_function_stats()->insert(functions_stats.begin(), functions_stats.end());
+  const FunctionInfoMap<FunctionStats>& functions_stats = capture_data.functions_stats();
+  for (const auto& function_to_stats : functions_stats) {
+    const FunctionInfo& function = function_to_stats.first;
+    const ModuleData* module = capture_data.FindModuleByPath(function.loaded_module_path());
+    CHECK(module != nullptr);
+    const ProcessData* process = capture_data.process();
+    CHECK(process != nullptr);
+    uint64_t absolute_address =
+        FunctionUtils::GetAbsoluteAddress(function_to_stats.first, *process, *module);
+    capture_info.mutable_function_stats()->operator[](absolute_address) = function_to_stats.second;
+  }
```

Comment:
Maybe consider having a method: `uint64_t CaptureData::GetAbsoluteAddress(const FunctionInfo&)`. Then `CaptureData::FindModuleByPath` is probably not needed

Reply:
Thanks for the suggestion. Done!